Writerside/topics/introduction.md

```diff
@@ -1,97 +1,82 @@
+research seeks to address these challenges while also considering scalability, as expanding transit systems further
+complicates the problem space.
+
+## Approach and Expected Outcomes
+
+This thesis adopts a comprehensive approach to address the identified challenges, beginning with the utilization of GTFS
+data as the foundation for the transit routing system. The standardized format of GTFS [3], which includes
+static information such as stop times, routes, and trips, provides a consistent and reliable dataset for the
+implementation of routing algorithms. A memory-efficient library for reading and processing GTFS schedules will be
+developed to enable scalable management of large datasets, ensuring the system can handle large-scale transit networks
+effectively.
+
+One of the core aspects of this thesis is the implementation of the RAPTOR algorithm [4], which is
+designed to efficiently compute transit routes by considering time dependencies such as schedules and transfer windows.
+The RAPTOR algorithm employs a round-based approach, where transit routes are organized into discrete rounds, each
+representing the number of trips needed. This structure simplifies the traversal of the network by enabling the
+algorithm to explore transit options in rounds and return pareto-optimal solutions, minimizing both arrival time and the
+number of transfers.
+
+The system will be complemented with an integration layer that manages both the routing algorithm and the schedule,
+while handling all necessary conversions between the two. This design maintains the independence of the schedule and the
+routing algorithm, enabling a modular architecture. The modularity allows for system components to be modified or
+replaced without affecting external dependencies, ensuring the system's adaptability and long-term maintainability.
+
+To facilitate interaction with the system, a REST API will be developed, allowing external applications to access the
+routing and scheduling services. This API will provide the foundation for seamless integration between the routing
+algorithm and external systems. As a proof of concept, a simple web application will be created to visualize search
+results and demonstrate the capabilities of the backend service. While the web application will serve as a proof of
+concept, the primary focus of this thesis remains on developing the backend service.
+
+This approach ensures the development of a scalable, high-performance public transit routing package based on the RAPTOR
+algorithm, as well as a flexible service component capable of managing complex transit schedules and routes. The end
+result is a modular system, complemented by a user-friendly interface that showcases the potential of the underlying
+backend service.
+
+## Scope and Delimitation
+
+This thesis is carried out as part of a student project, with the primary objective of advancing knowledge and
```

Comment:
Maybe a bit shorter, and also omit the intention of the commercial product, I think this is clear given the part `no external company interests`:

```
This thesis is part of the Master of Advanced Studies in Software Engineering at the Eastern Switzerland University of
Applied Sciences (OST). It focuses on applying the knowledge gained from the program to further advance methodologies in
public transit routing, with no external company interests influencing the scope or outcomes of this research.
```

Reply:
0a3dc1b